src/macaron/slsa_analyzer/checks/base_check.py

```diff
@@ -7,18 +7,22 @@
 from abc import abstractmethod
 
 from macaron.slsa_analyzer.analyze_context import AnalyzeContext
-from macaron.slsa_analyzer.checks.check_result import CheckResult, CheckResultType, SkippedInfo, get_result_as_bool
-from macaron.slsa_analyzer.slsa_req import ReqName, get_requirements_dict
+from macaron.slsa_analyzer.checks.check_result import (
+    CheckInfo,
+    CheckResult,
+    CheckResultData,
+    CheckResultType,
+    SkippedInfo,
+    get_result_as_bool,
+)
+from macaron.slsa_analyzer.slsa_req import ReqName
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
 class BaseCheck:
     """This abstract class is used to implement Checks in Macaron."""
 
-    # The dictionary that contains the data of all SLSA requirements.
-    SLSA_REQ_DATA = get_requirements_dict()
-
     def __init__(
```

Comment:
The docstring of this constructor needs updating.

Reply:
What needs updating? The parameters have not changed, and the docstring still seems correct to me.
Only thing I think we could add would be declarations for the class' fields, since they used to be 1-1 with the constructor parameters but now are not.

Reply:
Oh, sorry nvm. My mistake. I think it was decided that we don't document class fields in this case. ([Our style guide](https://oracle.github.io/macaron/pages/developers_guide/style_guide.html#docstrings-for-classes))

Reply:
In this case the class fields are not private, they are accessed by registry and runner, so according to the style guide I suppose they should be made into properties and should be documented.
They are also accessed by subclasses, and creating additional subclasses of the BaseCheck class is intended to be an extension point for Macaron, so in that sense they are part of the "public interface", and so it seems worthwhile documenting them for that reason alone (though if they were only intended to be accessible to subclasses and nothing else, the style guide doesn't specify whether we should do anything special for such a case of a "protected" field).

Reply:
That sounds reasonable.